projects/vdk-plugins/vdk-oracle/src/vdk/plugin/oracle/oracle_plugin.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+from typing import List
+
+import click
+import oracledb
+from tabulate import tabulate
+from vdk.api.plugin.hook_markers import hookimpl
+from vdk.api.plugin.plugin_registry import IPluginRegistry
+from vdk.internal.builtin_plugins.run.job_context import JobContext
+from vdk.internal.core.config import ConfigurationBuilder
+from vdk.internal.util.decorators import closing_noexcept_on_close
+from vdk.plugin.oracle.ingest_to_oracle import IngestToOracle
+from vdk.plugin.oracle.oracle_configuration import ORACLE_CONNECTION_STRING
+from vdk.plugin.oracle.oracle_configuration import ORACLE_PASSWORD
+from vdk.plugin.oracle.oracle_configuration import ORACLE_USER
+from vdk.plugin.oracle.oracle_configuration import OracleConfiguration
+from vdk.plugin.oracle.oracle_connection import OracleConnection
+
+"""
+Include the plugins implementation. For example:
+"""
+
+log = logging.getLogger(__name__)
+
+
+class OraclePlugin:
+    @hookimpl(tryfirst=True)
+    def vdk_configure(self, config_builder: ConfigurationBuilder):
+        OracleConfiguration.add_definitions(config_builder)
+
+    @hookimpl
+    def initialize_job(self, context: JobContext):
+        conf = OracleConfiguration(context.core_context.configuration)
+        context.connections.add_open_connection_factory_method(
+            "ORACLE",
+            lambda: OracleConnection(
+                conf.get_oracle_user(),
+                conf.get_oracle_password(),
+                conf.get_oracle_connection_string(),
+            ),
+        )
+        context.ingester.add_ingester_factory_method(
+            "oracle", (lambda: IngestToOracle(context.connections))
+        )
+
+
+@hookimpl
+def vdk_start(plugin_registry: IPluginRegistry, command_line_args: List):
+    plugin_registry.load_plugin_with_hooks_impl(OraclePlugin(), "OraclePlugin")
+
+
+@click.command(
+    name="oracle-query",
```

Comment:
Remove oracle-query . 

All <db-name>-query have been deprecated in favour of more generic `vdk sql-query` added in this PR https://github.com/vmware/versatile-data-kit/pull/2649

Reply:
I'd rather not touch it for now. I barely got everything working and don't want to spend more time on it. Here's a separate ticket to fix it and I'll also put that in a comment above the implementation, so we don't forget.

https://github.com/vmware/versatile-data-kit/issues/2940

Reply:
You need only remove the 'oracle-query' command and switch to `sql-query` in the test as in [vdk-duckdb tests] (https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-plugins/vdk-duckdb/tests/test_plugin.py#L70)

It shouldn't be more than a 15 minutes. Seems too minor for a whole ticket. But it's up to you.